src/main/java/net/neoforged/moddevgradle/tasks/JarJar.java

```diff
@@ -47,25 +51,67 @@ public abstract class JarJar extends DefaultTask {
+    protected void run() throws IOException {
+        List<ResolvedJarJarArtifact> includedJars = new ArrayList<>(getJarJarArtifacts().getResolvedArtifacts().get());
         fileSystemOperations.delete(spec -> spec.delete(getOutputDirectory()));
 
+        var artifactFiles = new ArrayList<>(includedJars.stream().map(ResolvedJarJarArtifact::getFile).toList());
+        // Now we have to handle pure file collection dependencies that do not have artifact ids
+        for (var file : getInputFiles()) {
+            if (!artifactFiles.contains(file)) {
+                // Determine the module-name of the file, which is also what Java will use as the unique key
+                // when it tries to load the file. No two files can have the same module name, so it seems
+                // like a fitting key for conflict resolution by JiJ.
+                var moduleName = FileUtils.getExplicitJavaModuleName(file);
+                if (moduleName.isEmpty()) {
+                    throw new GradleException("Cannot embed local file dependency " + file + " because it has no explicit Java module name.\n" +
+                                              "Please set either 'Automatic-Module-Name' in the Jar manifest, or make it an explicit Java module.\n" +
+                                              "This ensures that your file does not conflict with another mods library that has the same or a similar filename.");
+                }
+
+                // Create a hashcode to use as a version
+                var hashCode = FileUtils.hashFile(file, "MD5");
+                includedJars.add(new ResolvedJarJarArtifact(
+                        file,
+                        file.getName(),
+                        hashCode,
+                        "[" + hashCode + "]",
+                        "",
+                        moduleName.get()
+                ));
+                artifactFiles.add(file);
+            }
+        }
+
         // Only copy metadata if not empty, always delete
         if (!includedJars.isEmpty()) {
             fileSystemOperations.copy(spec -> {
                 spec.into(getOutputDirectory().dir("META-INF/jarjar"));
-                spec.from(includedJars.stream().map(ResolvedJarJarArtifact::getFile).toArray());
+                spec.from(artifactFiles.toArray());
                 for (var includedJar : includedJars) {
+                    // Warn if any included jar is using the cursemaven group.
+                    // We know that cursemaven versions are not comparable, and the same artifact might also be
+                    // available under a "normal" group and artifact from another Maven repository.
+                    // JIJ will not correctly detect the conflicting file at runtime if another mod uses the normal Maven dependency.
+                    // For a description of Curse Maven, see https://www.cursemaven.com/
+                    if ("curse.maven".equals(includedJar.getGroup())) {
```

Comment:
This check should also be introduced for the modrinth maven.

Reply:
https://support.modrinth.com/en/articles/8801191-modrinth-maven

The group is `maven.modrinth`.